lib/ansible/modules/source_control/github_organization.py

```diff
@@ -0,0 +1,308 @@
+    name: my-organization
+    owner: a-user
+
+- name: Rename an organization
+  github_organization:
+      state: rename
+      base_url: https://github-enterprise.domain.com
+      token: 'xxxxxxxxxxxxxxxxxxxxxxxxxxx'
+      token_user: token-username
+      name: my-organization
+      rename_to: my-renamed-organization
+'''
+
+import base64
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+
+
+def _build_api_url(base_url, path):
+    url = base_url
+
+    if url.endswith('/'):
+        if path.startswith('/'):
+            url += path[1:]
+        else:
+            url += path
+    else:
+        if path.startswith('/'):
+            url += path
+        else:
+            url += '/' + path
+
+    return url
+
+
+def _get_auth_header(user, token):
+    auth = base64.encodestring('%s:%s' % (user, token)).replace('\n', '')
+    return 'Basic %s' % auth
+
+
+def _get_organization(module, base_url, auth_header, name):
+    path = "orgs/%s" % (name)
+    url = _build_api_url(base_url, path)
+    headers = {'Authorization': auth_header}
+
+    r, info = fetch_url(module, url, method="GET", headers=headers)
+    if info['status'] == 200:
+        return module.from_json(r.read())
+    else:
+        return None
+
+
+def _get_base_url(module):
+    url = module.params['base_url']
+
+    if not url.endswith('api/v3'):
+        url = _build_api_url(url, 'api/v3')
+
+    return url
+
+
+def _merge_organization_result(res, org):
+    res.update({
+        'name': org['name'],
+        'id': org['id'],
+        'created_at': org['created_at'],
+        'updated_at': org['updated_at'],
+        'collaborators': org['collaborators'],
+        'public_repo_count': org['public_repos'],
+        'private_repo_count': org['total_private_repos'],
+        'default_repo_permission': org['default_repository_permission']
+    })
+    return res
+
+
+def create_organization(module, auth_header):
+    result = {'changed': False}
+
+    organization_name = module.params['name']
+    base_url = _get_base_url(module)
+
+    existing_org = _get_organization(module, base_url, auth_header, organization_name)
+
+    if existing_org is None:
+        url = _build_api_url(base_url, 'admin/organizations')
+        headers = {'Authorization': auth_header}
+        data = {
+            "login": organization_name,
+            # "profile_name": organization_name,
+            "admin": module.params['owner']
+        }
+
+        r, info = fetch_url(module,
+                            url,
+                            data=module.jsonify(data),
+                            method="POST",
+                            headers=headers)
+        if info['status'] == 201:
+            org = _get_organization(module, base_url, auth_header, organization_name)
+            result.update({'changed': True})
+            _merge_organization_result(result, org)
+        else:
+            failure = module.from_json(info['body'])
```

Comment:
Will info['body'] be json? Might be a string/HTML.

Reply:
The GitHub API should only return JSON here, it is documented as such. I can go more defensive with try/catch but it should only be returning JSON here.

Reply:
Someone could have a proxy in front, and if it intercepted a non-200 status code could replace the body with HTML or otherwise.

Reply:
Good point!